Could not build JoltPhysics 5.2.0 with Apple Clang 13: incomplete type 'JPH::Color'

Greetings!

First, thank you for providing this project to the community. I'm here not only to report a case but also to ask for guidance due the present build scenario:

We are trying to package Joltphysics 5.2.0 on [ConanCenterIndex](https://github.com/conan-io/conan-center-index/pull/24670) and we are facing an error when building it to Mac OSX, by using Clang 13:

```
In file included from /Applications/conan/xcode/13.0/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.3.sdk/usr/include/c++/v1/__tuple:15:
/Applications/conan/xcode/13.0/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.3.sdk/usr/include/c++/v1/type_traits:3559:38: error: incomplete type 'JPH::Color' used in type trait expression
    : public integral_constant<bool, __is_trivial(_Tp)>

...

/Users/jenkins/workspace/cci_prod_PR-24670/conan-home/p/b/joltpc9c3d9371d78d/b/src/Jolt/Core/Color.cpp:12:13: error: incomplete type 'JPH::Color' named in nested name specifier
const Color Color::sBlack(0, 0, 0);
            ^~~~~~~
/Users/jenkins/workspace/cci_prod_PR-24670/conan-home/p/b/joltpc9c3d9371d78d/b/src/Build/../Jolt/Core/Color.h:9:7: note: forward declaration of 'JPH::Color'
class Color;
      ^
/Users/jenkins/workspace/cci_prod_PR-24670/conan-home/p/b/joltpc9c3d9371d78d/b/src/Jolt/Core/Color.cpp:13:13: error: incomplete type 'JPH::Color' named in nested name specifier
const Color Color::sDarkRed(128, 0, 0);
            ^~~~~~~
/Users/jenkins/workspace/cci_prod_PR-24670/conan-home/p/b/joltpc9c3d9371d78d/b/src/Build/../Jolt/Core/Color.h:9:7: note: forward declaration of 'JPH::Color'
class Color;
      ^
/Users/jenkins/workspace/cci_prod_PR-24670/conan-home/p/b/joltpc9c3d9371d78d/b/src/Jolt/Core/Color.cpp:14:13: error: incomplete type 'JPH::Color' named in nested name specifier
const Color Color::sRed(255, 0, 0);
            ^~~~~~~
```

I know the compiler version is a bit old, but the README points Clang +10 compatible: https://github.com/jrouwe/JoltPhysics?tab=readme-ov-file#compiling

You can find the full build log [here](https://c3i.jfrog.io/artifactory/cci-build-logs/cci/prod/PR-24670/3/package_build_logs/build_log_joltphysics_5_2_0_7a6ec440933fbea9259f2479477df737_c61e9af79e5b573dcc59aa722ab481a49519393e.txt)

Our CI environment is the following:

```
CPU: armv8
OS: Mac OSX
Build Type: Release
Compiler: Apple Clang 13
Apple SDK: /Applications/conan/xcode/13.0/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.3.sdk
```

```
CMake options:

TARGET_UNIT_TESTS = OFF
TARGET_HELLO_WORLD = OFF
TARGET_PERFORMANCE_TEST = OFF
TARGET_SAMPLES = OFF
TARGET_VIEWER = OFF
CROSS_PLATFORM_DETERMINISTIC = OFF
INTERPROCEDURAL_OPTIMIZATION = OFF
GENERATE_DEBUG_SYMBOLS = OFF
ENABLE_ALL_WARNINGS = OFF
OVERRIDE_CXX_FLAGS = OFF
DEBUG_RENDERER_IN_DEBUG_AND_RELEASE = OFF
PROFILER_IN_DEBUG_AND_RELEASE = OFF
```

We downloaded the exact tag version and has no patches applied: https://github.com/jrouwe/JoltPhysics/archive/refs/tags/v5.2.0.tar.gz

---- 

The error is related to the compiler version most probably because I can build the same using Apple Clang 16 on Mac M1 (armv8) without errors. Here is my local build log with success: 

[joltphysics-5.2.0-osx-static-clang16.log](https://github.com/user-attachments/files/18577097/joltphysics-5.2.0-osx-static-clang16.log)


So I would like to ask if the project requires a newer Apple Clang version, or if it desires a patch by giving Apple Clang 13 support?

Regards!

--- a/Jolt/Core/Color.h
+++ b/Jolt/Core/Color.h
@@ -12,7 +12,7 @@ class Color;
 using ColorArg = Color;
 
 /// Class that holds an RGBA color with 8-bits per component
-class [[nodiscard]] JPH_EXPORT_GCC_BUG_WORKAROUND Color
+class JPH_EXPORT_GCC_BUG_WORKAROUND [[nodiscard]] Color
 {
 public:
 	/// Constructors

--- a/Jolt/Geometry/OrientedBox.h
+++ b/Jolt/Geometry/OrientedBox.h
@@ -14,7 +14,7 @@ JPH_NAMESPACE_BEGIN
 class AABox;
 
 /// Oriented box
-class [[nodiscard]] JPH_EXPORT_GCC_BUG_WORKAROUND OrientedBox
+class JPH_EXPORT_GCC_BUG_WORKAROUND [[nodiscard]] OrientedBox
 {
 public:
 	JPH_OVERRIDE_NEW_DELETE

--- a/Jolt/Physics/Body/Body.h
+++ b/Jolt/Physics/Body/Body.h
@@ -31,7 +31,7 @@ class SoftBodyCreationSettings;
 /// The functions that get/set the position of the body all indicate if they are relative to the center of mass or to the original position in which the shape was created.
 ///
 /// The linear velocity is also velocity of the center of mass, to correct for this: \f$VelocityCOM = Velocity - AngularVelocity \times ShapeCOM\f$.
-class alignas(JPH_RVECTOR_ALIGNMENT) JPH_EXPORT_GCC_BUG_WORKAROUND Body : public NonCopyable
+class JPH_EXPORT_GCC_BUG_WORKAROUND alignas(JPH_RVECTOR_ALIGNMENT) Body : public NonCopyable
 {
 public:
 	JPH_OVERRIDE_NEW_DELETE

